selfdrive/ui/qt/offroad/settings.cc

```diff
@@ -240,13 +241,23 @@ SoftwarePanel::SoftwarePanel(QWidget* parent) : QWidget(parent) {
     } else if (path.contains("LastUpdateTime")) {
       updateLabels();
     }
+    fs_watch->deleteLater();
+    fs_watch = nullptr;
   });
+  std::system("pkill -1 -f selfdrive.updated");
 }
 
 void SoftwarePanel::showEvent(QShowEvent *event) {
   updateLabels();
 }
 
+void SoftwarePanel::hideEvent(QHideEvent* event) {
+  if (fs_watch) {
```

Comment:
why not just remove and add the paths?

Reply:
I think delete is more clearer, or I have to save paths in the class variable, or write it here  again